script/libtest.sh

```diff
@@ -58,12 +58,11 @@ k8s_wait_for_job_completed() {
     while [ "$retryTimeSeconds" -gt 0 ]; do
         # Avoid to exit the function if the job is not completed yet
         set +e
-        kubectl get jobs -n $namespace -l $labelSelector \
-           -o jsonpath='{.items[*].status.conditions[?(@.type=="Complete")].status}' | grep "True"
+        kubectl get jobs -n $namespace -l $labelSelector -o jsonpath='{.items[*].status.conditions[?(@.type=="Complete")].status}' | grep "True"
         res=$?
         set -e
         # There is a job that finished
-        if [[ $res ]]; then
+        if [[ "$res" -eq "0" ]]; then
```

Comment:
This was the error:

```
▶ a=1; if [[ $a ]]; then echo foo; fi
foo

▶ a=0; if [[ $a ]]; then echo foo; fi
foo
```

Reply:
Thanks @andresmgot, that makes sense. 

I believe that the underlying issue is that we overrode the `e` option around `kubectl.... | grep true` so the script did not fail right? In any case we needed the `set +e` for some reason I do not remember.

Reply:
> In any case we needed the set +e for some reason I do not remember.

The command `kubectl get jobs` may fail because the job has not been created yet. That's why we need to "catch" errors there so we can retry.